cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -611,16 +613,34 @@ func (p *BicepProvider) Deploy(ctx context.Context) (*provisioning.DeployResult,
 		return nil, err
 	}
 
-	err = p.validatePreflight(
-		ctx,
-		bicepDeploymentData.Target,
-		bicepDeploymentData.CompiledBicep.RawArmTemplate,
-		bicepDeploymentData.CompiledBicep.Parameters,
-		deploymentTags,
-		optionsMap,
-	)
+	userConfig, err := p.userConfigManager.Load()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("loading user configuration: %w", err)
+	}
+
+	preflightDisableVar := "provision.disableValidation"
+	preflightDisable, exists := userConfig.GetString(preflightDisableVar)
+	if !(preflightDisable == "on" && exists) {
```

Comment:
Check `exists` first in the condition. If it `false` the second condition after && is not even evaluated.
Then, support any value for `preflightDisable` which is different then `off`, like `true`, `1`, `si`, `Y`, etc.
Optionally, re-use or copy this function: https://github.com/vhvb1989/azure-dev/blob/main/cli/azd/pkg/alpha/alpha_feature_manager.go#L139 to check if the setting is ON or OFF. 


```suggestion
	if exists && strings.ToLower(preflightDisable) != "off"  {
```